warehouse/warehouse.go

```diff
@@ -250,10 +252,14 @@ func (wh *HandleT) backendConfigSubscriber() {
 	ch := backendconfig.DefaultBackendConfig.Subscribe(context.TODO(), backendconfig.TopicBackendConfig)
 	for data := range ch {
 		wh.configSubscriberLock.Lock()
-		wh.warehouses = []warehouseutils.WarehouseT{}
+		wh.warehouses = []warehouseutils.Warehouse{}
 		config := data.Data.(map[string]backendconfig.ConfigT)
 		sourceIDsByWorkspaceLock.Lock()
 		sourceIDsByWorkspace = map[string][]string{}
+
+		wh.sourceIDToWorkspaceIDLock.Lock()
+		wh.sourceIDToWorkspaceID = map[string]string{}
```

Comment:
We are not populating sourceIDToWorkspaceID?

Reply:
Also, we already have sourceIDsByWorkspace map which contains workspaceID -> []sourceIDs. Should we rename this as it might be confusing now?

Reply:
Good catch; it was removed after an incorrect conflict resolution.

I will rename it to `workspaceBySourceID`